avocado/core/nrunner.py

```diff
@@ -359,6 +359,31 @@ class BaseRunner(abc.ABC):
     def __init__(self, runnable):
         self.runnable = runnable
 
+        # set default Avocado env variables
+        #
+        # NOTE: these environment variables are redefined inside an Avocado
+        # instrumented test.
+        #
+        # this is a code duplication from avocado/core/version.py to avoid
+        # nrunner depending on other than python modules
+        version = "unknown.unknown"
+        if PKG_RESOURCES_AVAILABLE:
+            try:
+                version = pkg_resources.get_distribution(
+                    'avocado-framework').version
+            except pkg_resources.DistributionNotFound:
+                pass
+        os.environ['AVOCADO_VERSION'] = version
+
+        # this is a code duplication from avocado/core/data_dir.py to avoid
+        # nrunner depending on other than python modules
+        os.environ['AVOCADO_TEST_BASEDIR'] = self.runnable.config.get(
+            'datadir.paths.base_dir', '')
+        os.environ['AVOCADO_TEST_LOGDIR'] = self.runnable.config.get(
+            'datadir.paths.logs_dir', '')
+        os.environ['AVOCADO_TEST_OUTPUTDIR'] = self.runnable.kwargs.get(
+            'AVOCADO_TEST_OUTPUT_DIR', '')
```

Comment:
Is this last line really needed ? I mean, all kwargs aren't being injected as variable envs?

Reply:
> Is this last line really needed ? I mean, all kwargs aren't being injected as variable envs?

Indeed, this line is not needed.

Reply:
What makes me think why we don't inject the others using the same method?

Reply:
My idea with this RFC is to discuss if the code is in the right place and if it makes sense.

Reply:
Maybe something like:

```python
def setup_env_variables(self):
  self.runnable.kwargs.update({'AVOCADO_TEST_LOGDIR': self.runnable.config.get('datadir.paths.logs_dir'),
                               'AVOCADO_TEST_BASEDIR':  self.runnable.config.get('datadir.paths.base_dir'))
  # And add here the code of `setup_output_dir()`
```